integration/test/test_power_balancer.py

```diff
@@ -0,0 +1,243 @@
+class TestIntegration_power_balancer(unittest.TestCase):
+
+    class AppConf(object):
+        """Class that is used by the test launcher in place of a
+        geopmpy.io.BenchConf when running the power_balancer benchmark.
+
+        """
+        def write(self):
+            pass
+
+        def get_exec_path(self):
+            script_dir = os.path.dirname(os.path.realpath(__file__))
+            return os.path.join(script_dir, '.libs', 'test_power_balancer')
+
+        def get_exec_args(self):
+            return []
+
+    @classmethod
+    def setUpClass(cls):
+        """Create launcher, execute benchmark and set up class variables.
+
+        """
+        cls._test_name = 'test_power_balancer'
+        cls._num_node = 4
+        cls._agent_list = ['power_governor', 'power_balancer']
+        # TODO: might want accessor methods?
+        # or rename to something like "test_config"
+        cls._skip_launch = util.g_util.skip_launch()
+        cls._show_details = util.g_util.show_details()
+        cls._tmp_files = []
+        cls._keep_files = (cls._skip_launch or
+                           os.getenv('GEOPM_KEEP_FILES') is not None)
+
+        # Clear out exception record for python 2 support
+        geopmpy.error.exc_clear()
+
+        if not cls._skip_launch:
+            num_rank = 16
+            loop_count = 500
+            fam, mod = geopm_test_launcher.get_platform()
+            power_budget = 200
+            if fam == 6 and mod == 87:
+                # budget for KNL
+                power_budget = 130
+            options = {'power_budget': power_budget}
+            gov_agent_conf_path = cls._test_name + '_gov_agent.config'
+            bal_agent_conf_path = cls._test_name + '_bal_agent.config'
+            cls._tmp_files.append(gov_agent_conf_path)
+            cls._tmp_files.append(bal_agent_conf_path)
+            path_dict = {'power_governor': gov_agent_conf_path, 'power_balancer': bal_agent_conf_path}
+
+            for app_name in ['geopmbench', 'socket_imbalance']:
+                app_conf = None
+                if app_name == 'geopmbench':
+                    app_conf = geopmpy.io.BenchConf(cls._test_name + '_app.config')
+                    cls._tmp_files.append(app_conf.get_path())
+                    app_conf.append_region('dgemm-imbalance', 8.0)
+                    app_conf.append_region('all2all', 0.05)
+                    app_conf.set_loop_count(loop_count)
+                    # Update app config with imbalance
+                    alloc_nodes = geopm_test_launcher.TestLauncher.get_alloc_nodes()
+                    for nn in range(len(alloc_nodes) // 2):
+                        app_conf.append_imbalance(alloc_nodes[nn], 0.5)
+                elif app_name == 'socket_imbalance':
+                    app_conf = cls.AppConf()
+                else:
+                    raise RuntimeError('No application config for app name {}'.format(app_name))
+                for agent in cls._agent_list:
```

Comment:
In the run scripts I've had on the big clusters, I've been putting a 60 second sleep between Governor and Balancer runs in the hope that it's enough time to cool the node.  Otherwise we may see thermal related throttling that impacts this test.

Reply:
I can add that here.  One of the done criteria for this story is to add an experiment type to abstract out this launch config.  The 60sec sleep should be in there too (probably between power caps too)